public/js/chord.js

```diff
@@ -0,0 +1,147 @@
+    .on("mouseout", fade(1));
+
+  var ticks = svg.append("g").selectAll("g")
+    .data(chord.groups)
+    .enter().append("g").selectAll("g")
+      .data(groupTicks)
+    .enter().append("g")
+      .attr("transform", function(d) {
+         return "rotate(" + (d.angle * 180 / Math.PI - 90) + ")"
+            + "translate(" + outerRadius + ",0)";
+    });
+
+  ticks.append("line")
+    .attr("x1", 1)
+    .attr("y1", 0)
+    .attr("x2", 5)
+    .attr("y2", 0)
+    .style("stroke", "#000");
+
+  ticks.append("text")
+    .attr("x", 8)
+    .attr("dy", ".35em")
+    .attr("transform", function(d) { return d.angle > Math.PI ? "rotate(180)translate(-16)" : null; })
+    .style("text-anchor", function(d) { return d.angle > Math.PI ? "end" : null; })
+    .text(function(d) { return d.label; });
+
+  svg.append("g")
+    .attr("class", "chord")
+    .selectAll("path")
+    .data(chord.chords)
+    .enter().append("path")
+    .attr("d", d3.svg.chord().radius(innerRadius))
+    .style("fill", function(d) { return fill(d.target.index); })
+    .style("opacity", 1);
+
+  // Returns an array of tick angles and labels, given a group.
+  function groupTicks(d) {
+    var k = (d.endAngle - d.startAngle) / 2
+    return [{
+      angle: k + d.startAngle,
+      label: members[d.index]
+    }];
+  }
+
+  // Returns an event handler for fading a given chord group.
+  function fade(opacity) {
+    return function(g, i) {
+      svg.selectAll(".chord path")
+        .filter(function(d) { return d.source.index != i && d.target.index != i; })
+        .transition()
+        .style("opacity", opacity);
+    };
+  }
+}
+
+function mArray(size) {
+    column = new Array();
+    for (var i=0; i<size; i++) {
+        column.push(new Array(size).fill(0));
+    }
+    return column;
+}
+
+
+var grouped = function(edges) {
+    return values = edges.reduce(function(p,c,ci,a) {
+        if (!p[c.sparkler]) { p[c.sparkler] = {} };
+        p[c.sparkler][c.sparklee] = c.weight;
+        return p;
+    },{});
+};
+
+var matrix = function(sparkles) {
+    var m2 = mArray(sparkles.nodes.length);
+    _.each(sparkles.nodes, function(sparkler,i) {
+        _.each(sparkles.nodes, function(sparklee,j) {
+            if (!sparkles.grouped[sparkler]) {
+            } else {
+                if (!sparkles.grouped[sparkler][sparklee]) {
+                } else {
+                    m2[i][j] = sparkles.grouped[sparkler][sparklee];
+                }
+            }
+        });
+    });
+    return m2;
+};
+
+$.get('/graph','',function(data) {
+    var all_nodes = function(){
+        var nodes = {}
+        $(data.edges).map(function(i,e) {
+            nodes[e.sparklee] = i;
+            nodes[e.sparkler] = i;
+        });
+        return Object.keys(nodes).sort();
+    };
+
+    data.nodes = [
+        "azizshamim",
+        "johnagan",
+        "gpadak",
+        "jessla",
+        "pmn",
+        "dsorkin",
+        "sachinr",
+        "davideg",
+        "joewadcan",
+    ];
```

Comment:
I don't think it matters much - since the full node set will be normalized to just the member set. Just need a way to inject the parameters in the node list. mostly i'd like validation that the members exist.